src/main/scala/wdl4s/values/WdlArray.scala

```diff
@@ -58,12 +58,12 @@ sealed abstract case class WdlArray(wdlType: WdlArrayType, value: Seq[WdlValue])
 
   def tsvSerialize: Try[String] = {
     wdlType.memberType match {
-      case t: WdlPrimitiveType => Success(value.map(_.valueString).mkString("\n"))
+      case t: WdlPrimitiveType => Success(value.map(_.valueString + "\n").mkString(""))
       case WdlObjectType => WdlObject.tsvSerializeArray(value map { _.asInstanceOf[WdlObject] })
```

Comment:
Yes, yes we should.  And we should have a test for that and the other cases here.